src/test/integration/IntegrationBase.t.sol

```diff
@@ -351,6 +347,234 @@ abstract contract IntegrationBase is IntegrationDeployer {
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable - delta, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                }
+            } else {
+                uint256 delta = prevAlloc.currentMagnitude - allocateParams.newMagnitudes[i];
+
+                // if deallocated, and completed
+                if (completed) {
+                    assertEq(
+                        curMagnitude.encumbered - delta, 
+                        prevMagnitude.encumbered, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable + delta, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                } else {
+                    assertEq(
+                        curMagnitude.encumbered, 
+                        prevMagnitude.encumbered, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                }
+            }
+
+            // Max magnitude should not have mutated.
+            assertEq(
+                curMagnitude.max, 
+                prevMagnitude.max, 
+                string.concat(err, " (max magnitude)")
+            );
+        }
+    }
+
+    function assert_Snap_Allocations_Slashed(
+        SlashingParams memory slashingParams,
+        OperatorSet memory operatorSet,
+        bool completed,
+        string memory err
+    ) internal {
+        User op = User(payable(slashingParams.operator));
+
+        Allocation[] memory curAllocs = _getAllocations(op, operatorSet, slashingParams.strategies);
+        Allocation[] memory prevAllocs = _getPrevAllocations(op, operatorSet, slashingParams.strategies);
+        Magnitudes[] memory curMagnitudes = _getMagnitudes(op, slashingParams.strategies);
+        Magnitudes[] memory prevMagnitudes = _getPrevMagnitudes(op, slashingParams.strategies);
+
+        (, uint32 delay) = allocationManager.getAllocationDelay(slashingParams.operator);
+
+        for (uint i = 0; i < slashingParams.strategies.length; i++) {
+            Allocation memory curAlloc = curAllocs[i];
+            Allocation memory prevAlloc = prevAllocs[i]; 
+
+            uint64 slashedMagnitude = uint64(uint256(prevAlloc.currentMagnitude).mulWadRoundUp(slashingParams.wadsToSlash[i]));
+
+            assertEq(
+                curAlloc.currentMagnitude, 
+                prevAlloc.currentMagnitude - slashedMagnitude, 
+                string.concat(err, " (currentMagnitude)")
+            );
+
+            if (completed) {
```

Comment:
Even if minimal, very much appreciate the comments done in 
`assert_Snap_Allocations_Modified` which indicate what block of code is checking which values like 
`// Max magnitude should not have mutated.`, `// Check allocations`, etc